### DISCUSSION TITLE
path resolve behavior

### DISCUSSION THREAD

**Discussion by youlinaa** [pnpm/pnpm#4635] [Q&A]
pnpm version 6.32.10

![image](https://user-images.githubusercontent.com/32128165/165665849-bc6e1cb1-c14a-4658-9a44-64138949fdf1.png)

moment is in dependencies, that strange.

**Comment by youlinaa** [comment]
![image](https://user-images.githubusercontent.com/32128165/165665206-c6880888-560a-4ffb-b4be-eb53ac1812f0.png)

but on my component the path is start with .pnpm

**Reply by zkochan** [reply]
You should have a symlink called moment in your node_modules. That symink should point to a directory inside `node_modules/.pnpm`.

**Reply by youlinaa** [reply]
There is no moment in node_modules, but there is in `node_modules/.pnpm`

![image](https://user-images.githubusercontent.com/32128165/166182687-9c865ccc-2880-4d66-b32a-a164d9e74a23.png)


Moment is in the dependencies of ant-design-vue, this is where I am confused, and other projects will not have this error message

Is this the cause of the monorepo?

![image](https://user-images.githubusercontent.com/32128165/166182390-d6d81ebb-5ad3-47eb-8266-dbcac07b5310.png)

**Reply by youlinaa** [reply]
Sorry, my mistake. other projects are configured with public-hoist-pattern.

![image](https://user-images.githubusercontent.com/32128165/166416732-327947d2-d814-45bb-bbe2-a8b5c5009ce2.png)

**Comment by larsivi** [comment]
See also https://github.com/pnpm/pnpm/discussions/4643

moment is a bad types package, so one possiblity is to add it to public-hoist-pattern

**Comment by youlinaa** [comment]
Actually, there are many such errors. 

It's a monorepo project, I don't know if it's related.

![image](https://user-images.githubusercontent.com/32128165/166181923-692e0f74-2409-457c-b32c-741cf8587bbc.png)

**Comment by youlinaa** [comment]
Solved, but why `vue-inbrowser-compiler-utils`

`.npmrc`

![image](https://user-images.githubusercontent.com/32128165/167776059-beba5889-71c9-4303-b245-b574a715fe8c.png)

`tsconfig`
![image](https://user-images.githubusercontent.com/32128165/167776090-d27cc3af-9ada-4e37-847e-41f1e6b004f2.png)
